crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -16,13 +17,28 @@ struct FragmentInput {
 
 @fragment
 fn fragment(in: FragmentInput) -> @location(0) vec4<f32> {
+    let is_orthographic = view.projection[3].w == 1.0;
+    let V = calculate_view(in.world_position, is_orthographic);
+    var uv = in.uv;
+#ifdef VERTEX_UVS
+#ifdef VERTEX_TANGENTS
+    if ((material.flags & STANDARD_MATERIAL_FLAGS_HEIGHT_MAP_BIT) != 0u) {
+        let tangent_V = vec3<f32>(
+            dot(V, in.world_tangent.xyz),
+            dot(V, -cross(in.world_normal, in.world_tangent.xyz) * sign(in.world_tangent.w)),
+            dot(V, in.world_normal),
+        );
```

Comment:
A clear case of "I don't know what I'm doing." Basically the idea is to convert the `V` (the view vector, aka `camera_pos - pixel_world_pos`) to the surface's tangent space. I simply copied the original code here, I feel like there might be a mathematical operation that expresses this more succinctly.